bdp_app/lib/pages/RoomsPage.dart

```diff
@@ -110,15 +118,29 @@ class _ChatListState extends State<ChatList> {
     });
   }
 
+
   void newMessage(chatMessage) {
-    // Create new list in map if conversation doesnt exist locally
-    if (!_messages.containsKey(chatMessage.chatID)) {
-      _messages[chatMessage.chatID] = new List<ChatMessage>();
+    //if message is not a group message
+    if(chatMessage.groupId == "") {
+      // Create new list in map if conversation doesnt exist locally
+      if (!_messages.containsKey(chatMessage.chatID)) {
+        _messages[chatMessage.chatID] = new List<ChatMessage>();
+      }
+      _messages[chatMessage.chatID].insert(
+          0,
+          chatMessage);
+      print("new" + _messages.toString());
+    }
+    //if message IS a group message
+    else{
+      if (!_messages.containsKey(chatMessage.groupId)) {
+        _messages[chatMessage.groupId] = new List<ChatMessage>();
+      }
+      _messages[chatMessage.groupId].insert(
+          0,
+          chatMessage);
+      print("new group message" + _messages.toString());
```

Comment:
Der Code ist in beiden Fällen identisch, der einzige Unterschied ist der key. Vorschlag: Mach sowas wie
```
var key = chatMessage.groupId == "" ? chatMessage.chatID : chatMessage.groupId;
``` 
Dann brauchst du später keine Fallunterscheidung mehr:
```
      if (!_messages.containsKey(key)) {
        _messages[key] = new List<ChatMessage>();
      }
      _messages[key].insert(
          0,
          chatMessage);
```

Reply:
schick :)

Reply:
schick-schick